Examples/Tests/ParticleDataPython/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,141 @@
+from pywarpx import picmi
+import numpy as np
+
+##########################
+# numerics parameters
+##########################
+
+dt = 7.5e-10
+
+# --- Nb time steps
+
+max_steps = 10
+
+# --- grid
+
+nx = 64
+ny = 64
+
+xmin = 0
+xmax = 0.03
+ymin = 0
+ymax = 0.03
+
+
+##########################
+# numerics components
+##########################
+
+grid = picmi.Cartesian2DGrid(
+    number_of_cells = [nx, ny],
+    lower_bound = [xmin, ymin],
+    upper_bound = [xmax, ymax],
+    lower_boundary_conditions = ['dirichlet', 'periodic'],
+    upper_boundary_conditions = ['dirichlet', 'periodic'],
+    lower_boundary_conditions_particles = ['absorbing', 'periodic'],
+    upper_boundary_conditions_particles = ['absorbing', 'periodic'],
+    moving_window_velocity = None,
+    warpx_max_grid_size = 32
+)
+
+solver = picmi.ElectrostaticSolver(
+    grid=grid, method='Multigrid', required_precision=1e-6,
+    warpx_self_fields_verbosity=0
+)
+
+##########################
+# physics components
+##########################
+
+electrons = picmi.Species(
+    particle_type='electron', name='electrons'
+)
+
+##########################
+# diagnostics
+##########################
+
+field_diag = picmi.FieldDiagnostic(
+    name = 'diag1',
+    grid = grid,
+    period = 10,
+    data_list = ['phi'],
+    write_dir = '.',
+    warpx_file_prefix = 'Python_particle_attr_access_plt'
+)
+
+##########################
+# simulation setup
+##########################
+
+sim = picmi.Simulation(
+    solver = solver,
+    time_step_size = dt,
+    max_steps = max_steps,
+    verbose = 1
+)
+
+sim.add_species(
+    electrons,
+    layout = picmi.GriddedLayout(
+        n_macroparticle_per_cell=[0, 0], grid=grid
+    )
+)
+sim.add_diagnostic(field_diag)
+
+sim.initialize_inputs()
+sim.initialize_warpx()
+
+##########################
+# python particle data access
+##########################
+
+from pywarpx import _libwarpx, callbacks
+
+_libwarpx.add_real_comp('electrons', 'newPid')
+
+def add_particles():
+
+    nps = 10
+    x = np.random.rand(nps) * 0.03
+    y = np.zeros(nps)
+    z = np.random.random(nps) * 0.03
+    ux = np.random.normal(loc=0, scale=1e4, size=nps)
+    uy = np.random.normal(loc=0, scale=1e4, size=nps)
+    uz = np.random.normal(loc=0, scale=1e4, size=nps)
+    attr = np.zeros((nps, 2))
+    attr[:,0] = 2.0
+    attr[:,1] = 5.0
+
+    _libwarpx.add_particles(
+        species_name='electrons', x=x, y=y, z=z, ux=ux, uy=uy, uz=uz,
+        attr=attr
```

Comment:
Should we potentially use a dictionary (key: string, value: np.array) here?
That way we could name the extra runtime attribute keys.

Reply:
Good idea! Thanks! The one downside I see with the new way is that there is now the extra overhead of always needing to sort the inputs in the appropriate order, but this is probably worth it.